src/components/Channel/Channel.tsx

```diff
@@ -1194,9 +1216,16 @@ const ChannelWithContext = <
     Re,
     Us
   >['retrySendMessage'] = async (message) => {
-    message = { ...message, status: 'sending' };
-    updateMessage(message);
-    await sendMessageRequest(message);
+    const messageWithoutReservedFields = removeReservedFields(message);
+
+    const statusPendingMessage = {
```

Comment:
Did I miss something or are these fields not just removed again when you call sendMessageRequest?